src/store/portfolioSummary/index.ts

```diff
@@ -0,0 +1,256 @@
+    const allClinList = await api.clinTable.getQuery(
+      {
+        params:
+          {
+            sysparm_fields: 'sys_id,clin_number,clin_status,funds_obligated',
+            sysparm_query: `sys_idIN` + clins
+          }
+      }
+    )
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records =
+          allClinList.filter(clin => (taskOrder.clins.indexOf(<string>clin.sys_id) !== -1));
+      })
+    })
+  }
+
+  /**
+   * Constructs a single query that gets all the cost records across all the clins. Parses
+   * the response and sets the costs to the respective clin.
+   */
+  @Action({rawError: true})
+  private async setCostsToTaskOrderClins(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    const clinNumbers: string[] = [];
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinNumbers.push(clinRecord.clin_number);
+        })
+      })
+    });
+    const allCostList = await api.costsTable.getQuery(
+      {
+        params:
+          {
+            sysparm_fields: 'sys_id,clin,task_order_number,is_actual,value',
+            sysparm_query: `clinIN` + clinNumbers
+          }
+      }
+    )
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinRecord.cost_records =
+            // @ts-ignore
+            allCostList.filter(cost => cost.clin === clinRecord.clin_number &&
+              cost.task_order_number === taskOrder.task_order_number); // FIXME temp code
+          // allCostList.filter(cost => cost.clin?.value === clinRecord.sys_id);//FIXME correct code
+        })
+      })
+    })
+  }
+
+  /**
+   * Performs the 'Total Obligated' and 'Funds Spent' for all the portfolios in the list
+   *
+   * NOTE: Instead of computing here if we were to make the aggregate API call, it would take
+   * as many calls as the number of portfolios to get, for example, the 'Total Obligated' from
+   * the CLINS table. The reason for this is, the aggregate query response does not have any
+   * reference to what portfolio the aggregate belongs to and so a call for each portfolio.
+   * Computing here will eliminate all these calls.
+   */
+  @Action({rawError: true})
+  private computeAllAggregations(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.dod_component = 'ARMY' // FIXME: delete this line after API starts returning
+      let totalObligatedForPortfolio = 0;
+      let totalFundsSpentForPortfolio = 0;
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          if (clinRecord.clin_status === 'ACTIVE' ||
+            clinRecord.clin_status === 'OPTION_EXERCISED') { // TODO: double check the statuses
```

Comment:
can this be double-checked for this PR or waiting for something coming down the road?

Reply:
This is correct as per my understanding during our meeting with Melissa. But wanted to leave it there as a quick note/ reminder. But will remove the "todo" comment if you want me to.